### PR TITLE
Internal CreateIndexRequest holds a single mapping

### PR DESCRIPTION
**Pull request by romseygeek**
Even though indexes can only have a single mapping, the internal
`CreateIndexRequest` objects return their mappings as a map keyed by
type.

This commit changes `CreateIndexRequest.mappings()` to return a simple
String, and alters the various mapping constructor methods to ensure that
this String is always in the correct format: an outer map keyed by `_doc`,
then the mappings with no further type information.  This is already handled
correctly on the REST layer.

Note that it does not make any further changes to `CreateIndexRequestBuilder`,
so as to limit the size of this changeset.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java [hunk 1]
```diff
@@ -211,6 +218,19 @@ public CreateIndexRequest settings(Map<String, ?> source) {
...
 
+    /**
+     * Set the mapping for this index
+     *
+     * The mapping should be in the form of a JSON string, with an outer _doc key
+     * <pre>
+     *     .mapping("{\"_doc\":{\"properties\": ... }}")
+     * </pre>
+     */
+    public CreateIndexRequest mapping(String mapping) {
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java]: One the client side [we are not nesting the mapping definition under the type](https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/indices/CreateIndexRequest.java#L156) . Are we ok with this difference?

Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java]: RestCreateIndexAction deals with adding this nesting in `prepareMappings`: https://github.com/elastic/elasticsearch/blob/37a997a9a960d8464185672ea56c76374991e896/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java#L68

**Review by mayya-sharipova — APPROVED** [review]
@romseygeek Thanks, the PR LGTM

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java]: It is, but I want to do this in a way that makes backports simple - just removing things will break more or less every backport to 7x from master.

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**